src/components/AddItem.js

```diff
@@ -1,22 +1,32 @@
 import React, { useState } from 'react';
 import { writeToFirestore } from '../lib/firebase';
 
-const AddItem = ({token}) => {
+const AddItem = ({ token, results }) => {
   let [name, setName] = useState('');
   let [frequency, setFrequency] = useState(7);
 
   function handleSubmitForm(event) {
     event.preventDefault();
-    if (name.length > 0) {
+    if (!name.trim()) {
+      alert(`Please enter a valid item`);
+    } else if (
+      results.filter(
+        result =>
+          name.replace(/[\W_]/g, '').toLowerCase() ===
+          result.name.replace(/[\W_]/g, '').toLowerCase(),
+      ).length
+    ) {
+      alert(`${name} already exists on your list`);
+    } else {
       writeToFirestore(token, {
         name,
         frequency,
         lastPurchaseDate: null,
       });
       alert(`${name} has been successfully added to your list.`);
-      setName('');
-      setFrequency(7);
     }
+    setName('');
+    setFrequency(7);
```

Comment:
It seems like any duplicate item with extra spaces never gets past the `if` and `else if` statements, but I noticed if you enter a new unique item, when it gets saved in the firestore, the extra spaces are still present. Might be worth saving `name.trim()` as the `name` state and using that throughout the handleSubmit function.

Reply:
After a user enters a duplicate item, we set the `name` back to being an empty string. Should we allow them to edit what they currently have? Or do you think that we should keep setting `name` back to an empty string? I'm not entirely sure if the use case I'm thinking of is too prevalent.

Reply:
Regex looks and works awesome!

Reply:
> It seems like any duplicate item with extra spaces never gets past the `if` and `else if` statements, but I noticed if you enter a new unique item, when it gets saved in the firestore, the extra spaces are still present. Might be worth saving `name.trim()` as the `name` state and using that throughout the handleSubmit function.

100% agree. Will update.

> After a user enters a duplicate item, we set the `name` back to being an empty string. Should we allow them to edit what they currently have? Or do you think that we should keep setting `name` back to an empty string? I'm not entirely sure if the use case I'm thinking of is too prevalent.

This is a great point. I think the best solution would be to include the feature outlined in issue #9 (included below) and only clear `name` when we receive a valid string or whitespace.

> - When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field